js/server/modules/@arangodb/aql-graph-traversal-generic-tests.js

```diff
@@ -237,6 +238,75 @@ const checkResIsValidGlobalBfsOf = (expectedVertices, actualPaths) => {
   assertEqual(expectedVertices, actualVertices, messages.join('; '));
 };
 
+const checkResIsValidShortestPath = (allowedPaths, actualPath) => {
+  let pathFound = false;
+
+  _.each(allowedPaths, function (currentPath) {
+    let innerKeyComparison = false;
+    // comparison only possible if lengths are equal, otherwise
+    // we do not have a hit
+    if (currentPath.length === actualPath.length) {
+      for (let i = 0; i < currentPath.length; i++) {
+        if (currentPath[i] === actualPath[i][0]) {
+          innerKeyComparison = true;
+        } else {
+          innerKeyComparison = false;
+          break;
+        }
+      }
+    }
+    if (innerKeyComparison) {
+      pathFound = true;
+    }
+  });
+
+  if (!pathFound) {
+    print("ShortestPath result not as expected!");
+    print("Allowed paths are: ");
+    print(allowedPaths);
+    print("Actual returned path is: ");
+    print(actualPath);
+  }
+
+  assertTrue(pathFound);
+};
+
+const checkResIsValidKShortestPath = (allowedPaths, actualPaths, expectedResults) => {
+  // check that we've only got as many paths as requested
+  if (actualPaths.length > expectedResults) {
+    print("Unexpected amount of found paths!");
+    print("Allowed paths are:");
+    print(allowedPaths);
+    print("Actual returned paths are: ");
+    print(actualPaths);
+  }
+  // we're allowed to find less or equal the amount of the set limit
+  assertTrue(actualPaths.length <= expectedResults);
+
+  if (actualPaths.length > 1) {
+    // also check that if we've received multiple paths, that each of them differ
+    // e.g.: [ [ "A", "B", "D", "F" ], [ "A", "C", "D", "F" ] ] <-- ALLOWED
+    // e.g.: [ [ "A", "B", "C", "D" ], [ "A", "B", "C", "D" ] ] <-- NOT ALLOWED
+    let currentArrayPosition = 0;
+    _.each(actualPaths, function (a) {
+      let clonedActualPaths = actualPaths.slice();
+      clonedActualPaths.splice(currentArrayPosition, 1);
+      _.each(clonedActualPaths, function (b) {
+        assertFalse(JSON.stringify(a) === JSON.stringify(b));
+      });
+      currentArrayPosition++;
+    });
+  }
+
+  _.each(actualPaths, function (actualPath) {
+    let path = [];
+    // format the result to be able to use already existing checkResIsValidShortestPath method
+    _.each(actualPath, function (vertex) {
+      path.push(vertex);
+    });
+    checkResIsValidShortestPath(allowedPaths, path);
+  });
+};
```

Comment:
This can be done for non-`EnabledWeightCheck` tests, where the number of vertic^H^Hexes determines the cost. The others version would require to have cost information at hand, wouldn't they?